source/basic.tex

```diff
@@ -389,6 +427,8 @@
 \end{itemize}
 
 \pnum
+A variable is named by an expression
+if the expression is an \grammarterm{id-expression} that denotes it.
 A variable \tcode{x} whose name appears as a
 potentially-evaluated expression $E$
```

Comment:
This and the following has 100% overlap with Davis' name lookup paper.

Given that fact, maybe we should prioritize applying the motions first before we dig into any medium-scale rewrites in this area.

Reply:
This is not an improvement.
Consider `sizeof(x) + 1`. Let's assume the entire expression is potentially-evaluated, so "x" is named by that expression (if you squint right). The phrasing "named as" clearly excludes the unevaluated operand, because that subexpression is not potentially-evaluated.

Reply:
Hmm. I was trying to fix the problem I see with "appears as", suggesting that the name *is* the expression.

Are either of these closer?:
"A variable x named in a potentially-evaluated"
or "name appears in a potentially-evaluated expression E"?

Reply:
No, "in" suggests/allows recursion that we don't want. And yes, in this case the name *is* the expression (it's a primary expression).